test/json.js

```diff
@@ -14,79 +14,140 @@
  */
 
 /* jslint node: true */
-/* global describe: false, before: false, after: false, it: false */
 'use strict';
 
-
-var NATS = require('../'),
+const NATS = require('../'),
     nsc = require('./support/nats_server_control'),
     should = require('should');
 
-describe('JSON payloads', function() {
-
-    var PORT = 1423;
-    var server;
+describe('JSON payloads', () => {
+    const PORT = 1423;
+    let server;
 
     // Start up our own nats-server
-    before(function(done) {
+    before((done) => {
         server = nsc.start_server(PORT, done);
     });
 
     // Shutdown our server
-    after(function(done) {
+    after((done) => {
         nsc.stop_server(server, done);
     });
 
+    function testPubSub({ input, expected }) {
+        if (expected === undefined) {
+            expected = input;
+        }
+
+        return (done) => {
+            const nc = NATS.connect({
+                json: true,
+                port: PORT
+            });
+
+            nc.subscribe('pubsub', (msg, reply, subj, sid) => {
+                if (msg instanceof Object) {
+                    msg.should.deepEqual(expected);
+                } else {
+                    should.strictEqual(msg, expected);
+                }
+                nc.unsubscribe(sid);
+                nc.close();
+
+                done();
+            });
+
+            nc.publish('pubsub', input);
+        };
+    }
+
+    function testReqRep({ input, expected, useOldRequestStyle }) {
```

Comment:
While we are not testing on node 4.5, our source is compatible and works correctly on node 4.5+. The object destructuring is wicked cool, but not supported in nodejs4.

Reply:
Maybe make it clear in the README then. Right now it says you're only targeting releases that are in maintenance or newer. Which is 6.x and newer.

I'll remove the destructuring.